src/controller/AutoCommissioner.cpp

```diff
@@ -48,23 +48,13 @@ void AutoCommissioner::SetOperationalCredentialsDelegate(OperationalCredentialsD
 CHIP_ERROR AutoCommissioner::SetCommissioningParameters(const CommissioningParameters & params)
 {
     mParams = params;
-    if (params.GetFailsafeTimerSeconds().HasValue())
-    {
-        ChipLogProgress(Controller, "Setting failsafe timer from parameters");
-        mParams.SetFailsafeTimerSeconds(params.GetFailsafeTimerSeconds().Value());
-    }
 
-    if (params.GetCASEFailsafeTimerSeconds().HasValue())
-    {
-        ChipLogProgress(Controller, "Setting CASE failsafe timer from parameters");
-        mParams.SetCASEFailsafeTimerSeconds(params.GetCASEFailsafeTimerSeconds().Value());
-    }
+    // Make sure any members that point to buffers that we are not pointing to
+    // our own buffers are not going to dangle.
+    mParams.ClearExternalBufferDependentValues();
```

Comment:
Will this be safe when a client calls AutoCommissioner.GetCommissioningParameters() updates a field and then calls AutoCommissioner.SetCommissioningParameters()?

For example, see:
https://github.com/project-chip/connectedhomeip/blob/c8a9f1a4906b3c97d01d6b9f35fc6f8cc34ec424/src/controller/java/CHIPDeviceController-JNI.cpp#L366

Reply:
So that specific case would actually work fine, I expect, because it's happening before any of the interesting state has been set on the commissioning params (which can happen in the middle of the commissioning process).

What would not work is doing that in the middle of commissioning, but I can add a check that would make it work.